app/src/test/java/org/dhis2/utils/granularsync/GranularSyncPresenterTest.kt

```diff
@@ -113,6 +117,133 @@ class GranularSyncPresenterTest {
         testSubscriber.assertValue(State.ERROR)
     }
 
+    @Test
+    fun `DataSet with ERROR completeRegistration should return ERROR from candidates`() {
+
+        whenever(
+            d2.dataSetModule().dataSetCompleteRegistrations()
+        ) doReturn mock()
+        whenever(
+            d2.dataSetModule().dataSetCompleteRegistrations()
+                .byDataSetUid()
+        ) doReturn mock()
+        whenever(
+            d2.dataSetModule().dataSetCompleteRegistrations()
+                .byDataSetUid().eq("data_set_uid")
+        ) doReturn mock()
```

Comment:
You can just join that lines using d2.dataSetModule().dataSetCompleteRegistrations()
                .byDataSetUid().eq("data_set_uid") in first time